app/src/main/java/com/github/leuludyha/ibdb/database/ActivityDao.kt

```diff
@@ -0,0 +1,29 @@
+package com.github.leuludyha.ibdb.database
+
+import androidx.lifecycle.LiveData
+import androidx.room.*
+
+/**
+ * DAOs is the interface between the Database and the user. It allows to facilitate operations
+ * on the database by abstracting the SQL.
+ */
+@Dao
+interface ActivityDao {
+    @Query("SELECT * FROM activity_table")
+    fun getAll(): LiveData<List<ActivityItem>>
+
+    @Query("SELECT * FROM activity_table ORDER BY RANDOM() LIMIT 1")
+    suspend fun getRandom(): ActivityItem?
+
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
+    suspend fun insert(activity: ActivityItem)
+
+    @Insert
+    fun insertAll(vararg activities: ActivityItem)
+
```

Comment:
Out of genuine curiosity, I understand why getAll is not declared suspend, since as I understand it LiveData is already asynchronous, but what is the reason for insertAll not to be so? As I understand it, it is asynchronous anyway through the call on WebApiActivityViewModel.kt, but why the difference?

Reply:
I agree for the tests hierarchy, I kind of rushed the tests and didn't take time for a good folders structure

About the `androidTest` and `test` folders, they're different. `androidTest` is for emulated tests (testing the api etc) and `test` is for the unit tests that don't require running the emulator, only backend is necessary.

I don't think we will merge this into main, I will rather recreate based on this code the correct database in the project.

About the suspend, I don't really know I just copied it from some samples. I'll look more in depth when implementing it in the real project.

I'll close this pull request now